mypy/typeanal.py

```diff
@@ -522,8 +523,8 @@ def bind_function_type_variables(self,
                 self.fail("Type variable '{}' is bound by an outer class".format(name), defn)
             self.tvar_scope.bind(name, tvar)
             binding = self.tvar_scope.get_binding(tvar.fullname())
-            assert binding is not None
-            defs.append(binding)
+            if binding is not None:
```

Comment:
Oops, sorry, this is a leftover from debugging the ``fullname`` problem, this change is not needed. Fixed.